build.gradle

```diff
@@ -192,10 +195,24 @@ dependencyUpdates.resolutionStrategy {
 
 // https://jeremylong.github.io/DependencyCheck/dependency-check-gradle/configuration.html
 dependencyCheck {
-  // Specifies if the build should be failed if a CVSS score above a specified level is identified.
-  // range of 7-10 fails the build, anything lower and it doesn't fail the build
-  failBuildOnCVSS = System.getProperty('dependencyCheck.failBuild') == 'true' ? 7.0 : 11
+  // We want any vulnerability, regardless of severity, to fail the build. The reason is that by failing the build we
+  //  have an opportunity to know the problem beforehand and open the related CVE ticket (and add temporary suppressions)
+  //  whereas if we don't fail the build we can know about potential problems only by tweaking the config or checking
+  //  the dependency-check reports, which may inadvertently fall through the cracks.
+  // If the restriction proves to be excessively severe, re-instate the following line
+  //  failBuildOnCVSS = System.getProperty('dependencyCheck.failBuild') == 'true' ? 7.0 : 11
```

Comment:
Please undo this. We dont want the build failing on every small CVE that comes up.

Reply:
The problem here is that CVEs need to be checked. If we don't fail the build, we don't have an opportunity to check them. My understanding is that in HMCTS all vulnerabilities and their related suppressions need to be screened.

Can we have a talk about this?

Reply:
Okay, so:
the original assumption that led to that flag being changed was that in HMCTS all vulnerabilities need to be screened.
After a conversation with Sean, it would appear that's not the case and they are interested only in High severity (7+) ones, meaning that we don't need to have 0 as a value.

I will now try to have a separate discussion around the logic using the environment variable (currently its absence implies that the build is not blocked, whereas we should probably be strict by default) and make the appropriate changes.